Update gpio-fan-overlay.dts

paulenuta opened the pull request:
Add references, links, clear details, some typo correction.
Paul
Signed-off-by: Paul Enuta paulenuta@gmail.com

paulenuta committed:
Update gpio-fan-overlay.dts

Add references, links, clear details, some typo correction.
Paul

pelwell commented on arch/arm/boot/dts/overlays/gpio-fan-overlay.dts:
```diff
@@ -1,29 +1,37 @@
- * sudo sh -c "echo '\n# Enable PI GPIO-Fan\ndtoverlay=gpio-fan\n' >> /boot/config.txt"
- * sudo sh -c "echo '\n# Enable PI GPIO-Fan\ndtoverlay=gpio-fan,gpiopin=12\n' >> /boot/config.txt"
+ * Build:
+ * 	- `sudo dtc -W no-unit_address_vs_reg -@ -I dts -O dtb -o /boot/overlays/gpio-fan.dtbo gpio-fan.dts`
```
The build instructions are incorrect because the filename is now `gpio-fan-overlay.dts`, but the kernel's `make dtbs` will do the compilation in most cases so this is less important.

pelwell commented on arch/arm/boot/dts/overlays/gpio-fan-overlay.dts:
```diff
@@ -1,29 +1,37 @@
+ * Activate:
+ *	- sudo nano /boot/config.txt add "dtoverlay=gpio-fan" or "dtoverlay=gpio-fan,gpiopin=12,temp=45000"
+ *	 or
+ *	- sudo printf "\n# Enable PI GPIO-Fan Default\ndtoverlay=gpio-fan\n" >> /boot/config.txt
```
This use of sudo doesn't work as written because the shell redirection runs as the current user. The correct form would be:
```
sudo sh -c 'printf "\n..." >> /boot/config.txt'
```
or
```
sudo sh -c "printf '\n...' >> /boot/config.txt"
```

paulenuta committed:
Update gpio-fan-overlay.dts

Correction of the overlay input file name "gpio-fan-overlay.dts" and of the activation commands, as rewied by @pelwell.

pelwell commented on arch/arm/boot/dts/overlays/gpio-fan-overlay.dts:
```diff
@@ -26,12 +26,12 @@
- *	- sudo printf "\n# Enable PI GPIO-Fan Default\ndtoverlay=gpio-fan\n" >> /boot/config.txt
- *	- sudo printf "\n# Enable PI GPIO-Fan Custom\ntoverlay=gpio-fan,gpiopin=12,temp=45000\n" >> /boot/config.txt"
+ *	- sudo sh -c 'printf "\n# Enable PI GPIO-Fan Default\ndtoverlay=gpio-fan\n" >> /boot/config.txt'
+ *	- sudo sh -c 'printf "\n# Enable PI GPIO-Fan Custom\ntoverlay=gpio-fan,gpiopin=12,temp=45000\n" >> /boot/config.txt"'
```
Close - there's a spurious extra `"` at the end.

paulenuta replied:
Right, missed that... :(

pelwell committed:
patch
